Centrifuge App: Wallet dialog

onnovisser opened the pull request:
<!-- This template is a starting point for creating a pull request. Not every pull request requires thorough documentation so use your best judgement. Typically, the higher the impact, the more documentation that is warranted. Feel free to remove sections that are not applicable to the pull request or use any combination of sections that make sense. -->

### Description

Adds the possibility to connect EVM wallets and adds a wallet dialog

### Approvals

<!-- Depending on the nature of the pull request, remove the roles that are not necessary for this review. Intricate technical changes may require two dev approvals. Reviewers should check the corresponding box after they approve. -->

- [ ] Dev

### Screenshots

<!-- Supporting screenshots, gifs, or videos to give reviewers an idea of how the pull request will affect the presentation of the app. -->

### Impact

<!-- Describe what parts of the app were affected so that reviewers can direct their focus. -->

Hornebom commented on centrifuge-react/src/utils/copyToClipboard.ts:
```diff
@@ -0,0 +1,13 @@
+export function copyToClipboard(value: string) {
```
I you feel tempted, check for support of the `navigator.clipboard` in the client
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard

offerijns commented:
I was testing this locally.  I really like where this is going, very clear flow that covers a lot of use cases (Polkadot and EVM wallets, proxied and non proxied accounts, different EVM networks, ...).

A few smaller points I noticed:
- Just naming, I think more consistent with the rest of the app would be: `Choose Account` => `Choose account`
- I would expect to be able to click on the Network / Account icons when viewing the account list, to be able to switch them. Ideal behaviour in my mind is that if I click on the wallet icon, I go back to the previous screen and the network is still selected but the wallet is unselected, and when I click on the network icon, both are unselected and I basically start from scratch.
- For some proxy types we have rewrites to phrase them better, e.g. `Any` => `Any rights`. Can we also implement this for `PermissionManagement` => `Manage permissions`, and any other missing proxy types similarly? (I’m not sure which ones are already implemented) 
- In the account list, there is always a `|` between the account name (e.g. Alice) and the address. In the designs this dash is only present for proxied accounts, i.e. it’s a dash between the proxied account and the proxy account (aka the delegator and delegatee). I think this is important as it’s now hard to differentiate proxied and non proxied accounts.
- The dropdown below the connect button seems to be narrower, in the designs in Figma it is as wide as the Connect button.

onnovisser commented:
Based on the above feedback


> Just naming, I think more consistent with the rest of the app would be: `Choose Account` => `Choose account`

Copy changed

> I would expect to be able to click on the Network / Account icons when viewing the account list, to be able to switch them. Ideal behaviour in my mind is that if I click on the wallet icon, I go back to the previous screen and the network is still selected but the wallet is unselected, and when I click on the network icon, both are unselected and I basically start from scratch.

Yes I think I would expect the same as a user. Made the icons clickable and go back to either wallet or network selection

> For some proxy types we have rewrites to phrase them better, e.g. `Any` => `Any rights`. Can we also implement this for `PermissionManagement` => `Manage permissions`, and any other missing proxy types similarly? (I’m not sure which ones are already implemented)

Added copy for PermissionManagement and PodAuth types

> In the account list, there is always a `|` between the account name (e.g. Alice) and the address. In the designs this dash is only present for proxied accounts, i.e. it’s a dash between the proxied account and the proxy account (aka the delegator and delegatee). I think this is important as it’s now hard to differentiate proxied and non proxied accounts.

Removed from non-proxies

> The dropdown below the connect button seems to be narrower, in the designs in Figma it is as wide as the Connect button.

Made it a bit wider but it's a fixed width, so won't always match the width of the button

offerijns commented:
Awesome! Checked again, only thing I still noticed is that if I click on the wallet, then it goes back to theCchoose wallet dialog with the wallet unselected at the top, but Polkadot.js still selected at the bottom, see:

<img width="618" alt="Screenshot 2023-02-20 at 18 46 35" src="https://user-images.githubusercontent.com/1748621/220172631-063f4024-d6d8-4a98-9e90-7450a7433d68.png">

Also, when I select Ttalisman which only has 1 connected account, the Switch wallet button in the dropdown is gone. I would expect it to still be shown in this case.

offerijns commented:
@onnovisser I tested again, everything above is now fixed! 💯

sophialittlejohn reviewed (approved):
:shipit:
